missive/emoji-mart
# chore: upgrade to babel v7

What this PR does:

- updates babel and its plugins/presets to v7 or equivalent
- updates jest to latest
- makes the `dist-modern` build use `@babel/env` with a list of latest Chrome/Edge/Firefox/Safari versions, rather than explicitly specifying every transform
- makes the legacy `dist` and `dist-es` builds use `@babel/env` with its default settings (i.e. run all ES2015 transforms)
- makes `@babel/runtime` an explicit `dependency` rather than `devDependency`
- removes most of the `src/polyfills`

I figure we are better off just making `@babel/runtime` into an explicit dependency rather than trying to write our own. Based on my testing, it adds a minimal amount of overhead (~4kB), and it's unlikely that our own polyfills will beat the built-in Babel ones. Plus it's just a lot of extra code we have to maintain.

Also, we no longer have to worry about bugs like #393 where a `core-js` dependency accidentally sneaks into our build and we break anyone who doesn't have this dependency.

Future TODOs:

- remove `_JSON`, it shouldn't be needed anymore (#402)
- make the `requestAnimationFrame` polyfill only run in Node (it's not needed in the browser) (#401)
